fix(selectfield): Issue #26 not displaying select icon

russell-tarpley opened the pull request:
A potential solution to Issue #26 
Leveraged code by @shaneog as well as a few other select implementations to arrive at a version that has been tested to work in IE11, Edge, Chrome, Firefox, Opera, & Safari

pointer-events: none was key to getting the clicking behavior to work correctly for the control
ms-expand: was key to getting IE 11 to render without it's native drop down

russell-tarpley committed:
Add Same Select Icon across browsers

Address Issue #26
https://github.com/mebibou/mdl-selectfield/issues/26

mebibou commented on src/selectfield/selectfield.scss:
```diff
@@ -50,17 +50,32 @@ $selectfield-error-color: unquote("rgb(222, 50, 38)") !default;
+  -ms-appearance: none;
   appearance: none;
 
+  &:-moz-focusring {
```
Can't find this weird pseudo-element anywhere, what is this?

russell-tarpley replied:
Firefox puts a dotted line around the select control when it gains focus by default. This eliminates that. I found it here [https://developer.mozilla.org/en-US/docs/Web/CSS/:-moz-focusring](url)

mebibou commented:
Looks like you branch is not up to date with the master branch, otherwise would only see 1 commit difference. Could you update it?

Also, there are too many different things in your PR. Please create one for the icon only, and you can create another one that fixes some other issues that would need to be tested more. Thanks

mebibou commented on package.json:
```diff
@@ -1,10 +1,10 @@
 {
-  "name": "mdl-selectfield",
+  "name": "package",
```
Please don't include this in any PR

mebibou commented on src/selectfield/selectfield.scss:
```diff
@@ -50,17 +50,32 @@ $selectfield-error-color: unquote("rgb(222, 50, 38)") !default;
     outline: none;
   }
 
+  &:focus::-ms-value {
```
Could you re-organize the order of the sub-elements so that you have all the peusdo-elements one after another, etc? it's easier to read

russell-tarpley replied:
Not a problem. I realized after I committed I had split them up. Good catch

mebibou commented on src/selectfield/selectfield.scss:
```diff
@@ -148,3 +163,20 @@ $selectfield-error-color: unquote("rgb(222, 50, 38)") !default;
+  cursor: pointer;
+  color: $selectfield-label-color;
+  left: 0;
+  padding: 2px 0;
```
Shouldn't this be set by a variable? it really depends on the height of the select element

russell-tarpley replied:
ah. the top padding. didn't see that at first. Could go back in and leverage a variable.

mebibou commented on src/selectfield/selectfield.js:
```diff
@@ -167,6 +167,11 @@
         this.select_.addEventListener('reset', this.boundResetHandler);
+        
+        var icon = document.createElement('div');
+        icon.className = 'mdl-selectfield__icon';
```
I am not a big fan of creating elements like this, this should work by adding the element in the html directly, something like:

``` html
<div class="mdl-selectfield mdl-js-selectfield">
  <select id="gender" class="mdl-selectfield__select">
    <option value=""></option>
    <option value="option1">option 1</option>
    <option value="option2">option 2</option>
  </select>
  <i class="material-icons">arrow_drop_down</i>
  <label class="mdl-selectfield__label" for="gender">User gender</label>
  <span class="mdl-selectfield__error">Select a value</span>
</div>
```

russell-tarpley replied:
Yea, this could go either way. I at least got rid of the need for an event handler. I like explicit markup just felt it would be redundant in this case. It's your component so I will make the adjustment.

russell-tarpley committed:
fix(scss) moved psuedo elements

grouped all pseudo elements together

russell-tarpley committed:
fix(package.json) revert changes

Reverted changes from previous commit

russell-tarpley committed:
fix(selectfield) remove dynamic div

Removed dynamic icon containing div from js file. To be replaced by div
included in the markup instead

russell-tarpley committed:
chore(build)

Re-built distribution files

russell-tarpley committed:
chore(documentation)

Updated README to reflect icon usage example

russell-tarpley commented:
I believe i got everything but updating the top padding for the icon div, which could be added in as a fix/enhancement. @mebibou Do you still want me to split this into multiple PRs? Also should I target your master or dev branch?

mebibou commented on package.json:
```diff
@@ -33,4 +33,4 @@
     "material-design-lite": "^1.1.3"
   }
-}
+}
```
Shouldn't be here either

mebibou commented on src/selectfield/selectfield.js:
```diff
@@ -167,7 +167,6 @@
         this.select_.addEventListener('focus', this.boundFocusHandler);
         this.select_.addEventListener('blur', this.boundBlurHandler);
         this.select_.addEventListener('reset', this.boundResetHandler);
-
```
Please do not commit useless changes

mebibou commented:
Well I can't test the `-ms` pseudo elements cos I am on mac, so not sure about pushing this to new release without anyone validating first. And yes always PR from master branch ;)

russell-tarpley committed:
Update mdl-selectfield.js

removed unnecessary change

russell-tarpley committed:
Update mdl-selectfield.js

unnecessary change removal

russell-tarpley commented:
Replaced this PR with #30 to clean up the commits and reduce the amount of change
